File: app/src/main/java/com/subrat/database/MedicineDao.java
package com.subrat.database;


import androidx.room.*;

import java.util.List;

/**
 * This is where we run queries for crud operation .
 */
@Dao
public interface MedicineDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    Long insert(MedicineEntity medicine);

    @Query("select * from medicines")
    List<MedicineEntity> getAllMedicines();

    @Query("select * from medicines where id =:medId")
    MedicineEntity getMedicineDetail(Long medId);

    @Update
    void updateMedicine(MedicineEntity medicine);


    @Query("delete from medicines where id=:medId")
    void deleteMedicine(Long medId);
}

File: app/src/main/java/com/subrat/database/MedicineDao.kt
package com.subrat.database


import androidx.room.*

/**
 * This is where we run queries for crud operation .
 */
@Dao
interface MedicineDao {

    @get:Query("select * from medicines")
    val allMedicines: List<MedicineEntity>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(medicine: MedicineEntity): Long?

    @Query("select * from medicines where id =:medId")
    fun getMedicineDetail(medId: Long): MedicineEntity

    @Update
    fun updateMedicine(medicine: MedicineEntity)


    @Query("delete from medicines where id=:medId")
    fun deleteMedicine(medId: Long)
}
